### PR TITLE
ThroughputBenchmark: improve formatting for ExecutionStats

### PR DESCRIPTION
**Pull request by salexspb**
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #22407 [easy] Add documentation file for torch.utils.throughput_benchmark&nbsp;&nbsp;[:yellow_heart:](https://our.internmc.facebook.com/intern/diff/D16078921/)
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#22293 ThroughputBenchmark: improve formatting for ExecutionStats**&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D16023999/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #22292 PyTorch ThroughputBenchmark: fix inaccuracy in number of iterations reporting&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D16023963/)

Just wrapping C class with nicer python interface which now
ust print dirrectly to get all the data. Later we can add various
visualizations there

Differential Revision: [D16023999](https://our.internmc.facebook.com/intern/diff/D16023999/)

### CODE REVIEW

File: torch/utils/throughput_benchmark.py [hunk 1]
```diff
@@ -2,6 +2,61 @@
...
+def format_time(time_us=None, time_ms=None, time_s=None):
+    '''Defines how to format time'''
+    assert sum([time_us is not None, time_ms is not None, time_s is not None]) == 1
+    if time_us is None:
+        if time_ms is not None:
+            time_us = time_ms * 1000.0
+        elif time_s is not None:
+            time_us = time_s * 1000.0 * 1000.0
+        else:
+            assert "Incorrect code :)"
```
Comment (dzhulgakov) [inline torch/utils/throughput_benchmark.py]: assert False?

File: torch/utils/throughput_benchmark.py [hunk 2]
```diff
@@ -2,6 +2,61 @@
...
+
+    @property
+    def iters_per_second(self):
+        '''
+        Returns total number of iterations per second across all calling threads
+        '''
+        return self.num_iters / self.total_time_seconds
+
+    @property
+    def total_time_seconds(self):
```
Comment (dzhulgakov) [inline torch/utils/throughput_benchmark.py]: alternative approach would have been to just assign these methods on the class, e.g.:

```
def total_time_seconds(self):
  ...

torch._C.ExecutionStats.total_time_seconds = property(total_time_seconds)
```

That way you don't need to create a wrapper class. But I don't insist - either works.

**Review by dzhulgakov — APPROVED** [review]

File: torch/utils/throughput_benchmark.py (hunk 2 above)
Comment (salexspb) [reply torch/utils/throughput_benchmark.py]: Hm, that would be quite a bit of injection of extra methods, including __str__. I personally find it a bit more readable with an explicit wrapper class.

File: torch/utils/throughput_benchmark.py (hunk 1 above)
Comment (jamesr66a) [reply torch/utils/throughput_benchmark.py]: assert False breaks lint :'( https://travis-ci.org/pytorch/pytorch/jobs/553100305

**Pull request closed without merging**